src/Integration.Vsix/CFamily/VcxProject/VcxRequestFactory.cs

```diff
@@ -75,36 +75,21 @@ internal class VcxRequestFactory : IRequestFactory
 
         public async Task<IRequest> TryCreateAsync(string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions)
         {
-            IRequest request = null;
-            await threadHandling.RunOnUIThread(() => request = TryCreateSync(analyzedFilePath, analyzerOptions));
```

Comment:
Previously everything related to VcxRequest creation was running on the UI thread, including the effective rules calculation, hence why the UI was locking up (#2784).
Only the FileConfig creation needs to be done on the UI thread as it uses the DTE object, so I've moved everything else to the background thread.
I think in general we should take the "closest to source" approach and put UI switching next to the code that really needs it, similar to what the VS analyzers suggest to do. Otherwise it can lead to situations like this one, where we introduce a UI lockup due to code that should've been running on the background thread.